frontend/src/tests/application-tests.js

```diff
@@ -1,3 +1,74 @@
 // TODO: Remove eslint disable
 // eslint-disable-next-line
-export function applicationsTests({ apiGET, apiPOST }) {}
+import PropTypes from "prop-types";
+import {
+    applicationPropTypes,
+    expect,
+    beforeAll,
+    it,
+    checkPropTypes
+} from "./utils";
+import { databaseSeeder } from "./setup";
+/**
+ * Tests for the API. These are encapsulated in a function so that
+ * different `apiGET` and `apiPOST` functions can be passed in. For example,
+ * they may be functions that make actual requests via http or they may
+ * be from the mock API.
+ *
+ * @param {object} api
+ * @param {Function} api.apiGET A function that when passed a route will return the get response
+ * @param {Function} api.apiPOST A function that when passed a route and data, will return the post response
+ */
+export function applicationsTests(api) {
+    const { apiGET, apiPOST } = api;
+    let session = null,
+        applicant = null;
+
+    const newApplicationData = {
+        comments: "",
+        program: "Phd",
+        department: "Computer Science",
+        previous_uoft_ta_experience: "Last year I TAed a bunch",
+        yip: 2,
+        annotation: "",
+        position_preferences: [
+            {
+                preference_level: 2,
+                position_id: 10
+            },
+            {
+                preference_level: 3,
+                position_id: 15
+            }
+        ]
+    };
+
+    // set up a session to be available before tests run
+    beforeAll(async () => {
+        await apiPOST("/debug/restore_snapshot");
+        session = databaseSeeder.seededData.session;
+        applicant = databaseSeeder.seededData.applicant;
+        newApplicationData.session_id = session.id;
```

Comment:
We don't have a style guide for this, but we have been using 
```
newApplicationData = {
    ...newApplicationData,
    session_id: session.id,
    applicant_id: applicant.id
}
```
Personally, I prefer doing it this way since `session_id` and `applicant_id` fields do not exist in `newApplicationData`. But it is just an suggestion, maybe we should come up with a consistent way of doing this.